audit/repositories/postgres_test.go

```diff
@@ -2,96 +2,96 @@ package repositories_test
 
 import (
 	"context"
-	"database/sql"
-	"errors"
-	"regexp"
 	"testing"
+	"time"
 
-	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/jmoiron/sqlx/types"
 	"github.com/odpf/salt/audit"
 	"github.com/odpf/salt/audit/repositories"
+	"github.com/odpf/salt/log"
+	"github.com/ory/dockertest/v3"
 	"github.com/stretchr/testify/suite"
-	"gorm.io/driver/postgres"
-	"gorm.io/gorm"
 )
 
 type PostgresRepositoryTestSuite struct {
 	suite.Suite
 
-	dbMock     sqlmock.Sqlmock
-	dbConn     *sql.DB
-	repository *repositories.PostgresRepository
+	repository     *repositories.PostgresRepository
+	pool           *dockertest.Pool
+	dockerResource *dockertest.Resource
 }
 
 func TestPostgresRepository(t *testing.T) {
 	suite.Run(t, new(PostgresRepositoryTestSuite))
 }
 
-func (s *PostgresRepositoryTestSuite) setupTest() {
-	db, dbMock, err := sqlmock.New()
-	s.Require().NoError(err)
-	s.dbConn = db
-	s.dbMock = dbMock
-	gormDB, err := gorm.Open(postgres.New(postgres.Config{
-		Conn: db,
-	}), &gorm.Config{})
-	s.Require().NoError(err)
-
-	s.repository = repositories.NewPostgresRepository(gormDB)
+func (s *PostgresRepositoryTestSuite) SetupSuite() {
+	var err error
+	s.repository, s.pool, s.dockerResource, err = newTestRepository(log.NewLogrus())
+	if err != nil {
+		s.T().Fatal(err)
+	}
 }
 
-func (s *PostgresRepositoryTestSuite) cleanupTest() {
-	s.dbConn.Close()
+func (s *PostgresRepositoryTestSuite) TearDownSuite() {
+	if err := s.repository.DB().Close(); err != nil {
+		s.T().Fatal(err)
+	}
+	if err := purgeTestDocker(s.pool, s.dockerResource); err != nil {
+		s.T().Fatal(err)
+	}
 }
 
 func (s *PostgresRepositoryTestSuite) TestInit() {
```

Comment:
Is this test needed? We aren't calling `Init` inside it since it is already called during `SetupSuite` and the test name is a bit confusing. And the same select statement is being run in other tests as well.

Reply:
removed, we can remove it now since we're already using dockertest